### DISCUSSION TITLE
With mongo db, how can I write model field to allow String or numeric values in the same field?

### DISCUSSION THREAD

**Discussion by realtebo** [prisma/prisma#23590] [Q&A] [answered]
### Question

I am connecting to a 3rd party datasource. 

a particular field could have numeric values, or string values. Same field, same collection, only different records as different data type in the same field.

How can i handle this situation in `schema.prisma` ?

**Comment by ludralph** [comment, marked as answer]
Hi @realtebo 👋 

Currently, Prisma does not support multiple data types for a single field in the schema. Each field in the Prisma schema must have a specific type defined. This is because Prisma Client generates types based on your schema and expects consistency in the data types.

However, there are a few workarounds you can consider:

1. Use the JSON data type: Prisma supports the JSON data type which can hold any valid JSON value. This allows you to store either a string or a number in the same field. However, Prisma Client will not provide type safety for the data inside the JSON field, and you will need to handle the data manually in your application code. For rexample:
```prisma
model YourModel {
  id    String @id @default(auto()) @map("_id") @db.ObjectId
  field Json
}
```

2. Data validation and transformation in your application code: You could define the field as a string in your Prisma schema, and then handle the conversion between numbers and strings in your application code. This would involve checking the type of the data before you send it to or retrieve it from the database, and converting it if necessary.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏
